src/colvarcomp.cpp

```diff
@@ -745,11 +745,18 @@ void colvar::cvc::debug_gradients()
   for (size_t ig = 0; ig < atom_groups.size(); ig++) {
     cvm::atom_group *group = atom_groups[ig];
     if (group->b_dummy) continue;
+    // Clear the gradients, because some CVCs may calculate the gradients in calc_value()
+    for (size_t ia = 0; ia < group->size(); ia++) {
+      group->grad_x(ia) = 0;
+      group->grad_y(ia) = 0;
+      group->grad_z(ia) = 0;
+    }
```

Comment:
Is this necessary if both `calc_value` and `calc_gradients` are called next? I'd expect the gradients to be overwritten in all cases.

Reply:
Not really. The code of some CVCs accumulate the gradients instead of overwriting them, like:
https://github.com/Colvars/colvars/blob/5a7978a8f4d1a0d087e68a612614ed04f2ebc251/src/colvarcomp_distances.cpp#L473-L479
and
https://github.com/Colvars/colvars/blob/5a7978a8f4d1a0d087e68a612614ed04f2ebc251/src/colvarcomp_coordnums.cpp#L98-L103
These code expect that the gradients are already cleared by `reset_atoms_data()`, and if they are called in `calc_value()`, then the gradients would be wrong in case of `debugGradients`, so it is necessary to clear the gradients and run the calculations again by `calc_value()` and then `calc_gradients()`.